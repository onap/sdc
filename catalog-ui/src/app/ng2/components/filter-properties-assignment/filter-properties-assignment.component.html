<popover-content #filterPopover [title]="'Filters'" [buttons]="footerButtons" placement="bottom-right" [hideArrow]="true">
    <!--<form [formGroup]="filterForm">-->
    <form>
        <div class="field">
            <label>Resource Type</label>
            <div>
                <checkbox [label]="'All'" [(checked)]="allSelected" (checkedChange)="selectAll()"></checkbox>
            </div>
            <div *ngFor="let type of typesOptions">
                <checkbox [label]="type" [(checked)]="selectedTypes[type]" (checkedChange)="onTypeSelected(type)"></checkbox>
            </div>
        </div>
        <div class="field">
            <label>Property Name</label>
            <input class="i-sdc-form-input"
                   name="propertyName"
                   [(ngModel)]="filterData.propertyName"
                   placeholder="Type here"
                   required
            />
        </div>
    </form>
</popover-content>
<div class="open-filter-button" [popover]="filterPopover" [ngClass]="{'open':showPopover}" (onShown)="showPopover = true" (onHidden)="showPopover = false">
    <div class="sprite-new filter-icon"></div>
</div>
