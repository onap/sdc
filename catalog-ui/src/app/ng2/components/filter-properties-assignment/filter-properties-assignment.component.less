@import '../../../../assets/styles/variables';
form{
    margin: 0 20px;
    .field{
        padding:20px 0;
        &:not(:last-child){
            border-bottom: solid 1px @main_color_o;
        }
        input{
            &::-webkit-input-placeholder { font-style: italic; } /* Safari, Chrome and Opera */
            &:-moz-placeholder { font-style: italic; } /* Firefox 18- */
            &::-moz-placeholder { font-style: italic; } /* Firefox 19+ */
            &:-ms-input-placeholder { font-style: italic; } /* IE 10+ */
            &:-ms-input-placeholder { font-style: italic; } /* Edge */
        }
    }
    /deep/ [ng-reflect-checked="true"]{
        /deep/ .checkbox-label-content{
            color: @main_color_a;
        }
    }
}

.open-filter-button{
    cursor: pointer;
    width: 32px;
    height: 34px;
    margin-left:5px;

    &.open{
        z-index: 1061;
        background-color: @main_color_p;
        border: solid 1px @main_color_c;
        border-bottom: none;
    }
    .filter-icon{
        top: 8px;
        right: 2px;
        position: relative;
    }
}

/deep/ .popover{
    border: solid 1px @main_color_c !important;
}
