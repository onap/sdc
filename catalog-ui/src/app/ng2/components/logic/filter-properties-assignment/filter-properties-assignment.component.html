<!--
 * Copyright (C) 2018 Huawei Intellectual Property. All rights reserved.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 -->
 
 
<popover-content #filterPopover [title]="'Filters'" [buttons]="footerButtons" placement="bottom-right" [hideArrow]="true">
    <!--<form [formGroup]="filterForm">-->
    <form>
        <div class="field">
            <label>Resource Type</label>
            <div>
                <checkbox [label]="'All'" [(checked)]="allSelected" (checkedChange)="selectAll()" data-tests-id="filter-checkbox-all"></checkbox>
            </div>
            <div *ngFor="let type of typesOptions">
                <checkbox [label]="type" [(checked)]="selectedTypes[type]" (checkedChange)="onTypeSelected(type)" [attr.data-tests-id]="'filter-checkbox-' + type.toLowerCase()"></checkbox>
            </div>
        </div>
        <div class="field">
            <label>Property Name</label>
            <input class="i-sdc-form-input"
                   name="propertyName"
                   [(ngModel)]="filterData.propertyName"
                   placeholder="Type here"
                   required
                   data-tests-id="filter-box"
            />
        </div>
    </form>
</popover-content>
<div class="open-filter-button" [popover]="filterPopover" >
    <div class="sprite-new filter-icon" data-tests-id="filter-button"></div>
</div>
