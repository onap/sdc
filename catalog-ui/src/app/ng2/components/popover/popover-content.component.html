<div #popoverDiv class="popover {{ effectivePlacement }}"
     [style.top]="top + 'px'"
     [style.left]="left + 'px'"
     [class.in]="isIn"
     [class.fade]="animation"
     style="display: block"
     role="popover"
     [ngClass]="{'hide-arrow':hideArrow}">
    <div [hidden]="!closeOnMouseOutside" class="virtual-area"></div>
    <div class="arrow" *ngIf="!hideArrow"></div>
    <div class="popover-header">
        <span class="title">{{ title }}</span>
        <span class="close-button" (click)="popover.hide()"></span>
    </div>
    <ng-content></ng-content>
    <div class="popover-footer">
        <button *ngFor="let buttonName of buttonsNames"
                class="tlv-btn {{buttons[buttonName].cssClass}}"
                [disabled] = "buttons[buttonName].getDisabled && buttons[buttonName].getDisabled()"
                (click) = "buttons[buttonName].callback()">{{buttons[buttonName].text}}</button>
    </div>
</div>


