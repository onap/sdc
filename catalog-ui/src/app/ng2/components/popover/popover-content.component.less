@import '../../../../assets/styles/variables';
@import '../../../../assets/styles/mixins';
@import '../../../../assets/styles/sprite-old';
.popover .virtual-area {
    height: 11px;
    width: 100%;
    position: absolute;
}
.popover.top .virtual-area {
    bottom: -11px;
}
.popover.bottom .virtual-area {
    top: -11px;
}
.popover.left .virtual-area {
    right: -11px;
}
.popover.right .virtual-area {
    left: -11px;
}
.popover.hide-arrow{
    margin: 0;
}

.popover-header{
    .m_14_m;
    font-weight: bold;
    -webkit-box-flex: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    height: 40px;
    line-height: 48px;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    text-align: left;
    border-bottom: solid 1px @main_color_o;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    margin: 0px 20px;
    max-width:350px;

    .title{
        flex: 1;
        max-width: 100%;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
    }
    .close-button{
        .sprite;
        .sprite.x-btn-black;
        cursor: pointer;
    }
}

.popover-footer{
    background-color: @tlv_color_t;
    height: 40px;
    clear: both;
    -webkit-box-flex: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: end;
    -ms-flex-pack: end;
    justify-content: flex-end;
    button{
        margin: 8px 12px 8px 6px;
    }
}
