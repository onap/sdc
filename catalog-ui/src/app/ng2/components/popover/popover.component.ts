/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

import { Directive, HostListener, ComponentRef, ViewContainerRef, ComponentFactoryResolver, ComponentFactory, Input, OnChanges, SimpleChange, Output, EventEmitter } from "@angular/core";
import {PopoverContentComponent} from "./popover-content.component";

@Directive({
    selector: "[popover]",
    exportAs: "popover"
})
export class PopoverComponent implements OnChanges {

    protected PopoverComponent = PopoverContentComponent;
    protected popover: ComponentRef<PopoverContentComponent>;
    protected visible: boolean;


    constructor(protected viewContainerRef: ViewContainerRef,
                protected resolver: ComponentFactoryResolver) {
    }

    @Input("popover")
    content: string|PopoverContentComponent;

    @Input()
    popoverDisabled: boolean;

    @Input()
    popoverAnimation: boolean;

    @Input()
    popoverPlacement: "top"|"bottom"|"left"|"right"|"auto"|"auto top"|"auto bottom"|"auto left"|"auto right";

    @Input()
    popoverTitle: string;

    @Input()
    popoverOnHover: boolean = false;

    @Input()
    popoverCloseOnClickOutside: boolean;

    @Input()
    popoverCloseOnMouseOutside: boolean;

    @Input()
    popoverDismissTimeout: number = 0;

    @Output()
    onShown = new EventEmitter<PopoverComponent>();

    @Output()
    onHidden = new EventEmitter<PopoverComponent>();

    @HostListener("click")
    showOrHideOnClick(): void {
        if (this.popoverOnHover) return;
        if (this.popoverDisabled) return;
        this.toggle();
    }

    @HostListener("focusin")
    @HostListener("mouseenter")
    showOnHover(): void {
        if (!this.popoverOnHover) return;
        if (this.popoverDisabled) return;
        this.show();
    }

    @HostListener("focusout")
    @HostListener("mouseleave")
    hideOnHover(): void {
        if (this.popoverCloseOnMouseOutside) return;
        if (!this.popoverOnHover) return;
        if (this.popoverDisabled) return;
        this.hide();
    }

    ngOnChanges(changes: {[propertyName: string]: SimpleChange}) {
        if (changes["popoverDisabled"]) {
            if (changes["popoverDisabled"].currentValue) {
                this.hide();
            }
        }
    }

    toggle() {
        if (!this.visible) {
            this.show();
        } else {
            this.hide();
        }
    }

    show() {
        if (this.visible) return;

        this.visible = true;
        if (typeof this.content === "string") {
            const factory = this.resolver.resolveComponentFactory(this.PopoverComponent);
            if (!this.visible)
                return;

            this.popover = this.viewContainerRef.createComponent(factory);
            const popover = this.popover.instance as PopoverContentComponent;
            popover.popover = this;
            popover.content = this.content as string;
            if (this.popoverPlacement !== undefined)
                popover.placement = this.popoverPlacement;
            if (this.popoverAnimation !== undefined)
                popover.animation = this.popoverAnimation;
            if (this.popoverTitle !== undefined)
                popover.title = this.popoverTitle;
            if (this.popoverCloseOnClickOutside !== undefined)
                popover.closeOnClickOutside = this.popoverCloseOnClickOutside;
            if (this.popoverCloseOnMouseOutside !== undefined)
                popover.closeOnMouseOutside = this.popoverCloseOnMouseOutside;

            popover.onCloseFromOutside.subscribe(() => this.hide());
            if (this.popoverDismissTimeout > 0)
                setTimeout(() => this.hide(), this.popoverDismissTimeout);
        } else {
            const popover = this.content as PopoverContentComponent;
            popover.popover = this;
            if (this.popoverPlacement !== undefined)
                popover.placement = this.popoverPlacement;
            if (this.popoverAnimation !== undefined)
                popover.animation = this.popoverAnimation;
            if (this.popoverTitle !== undefined)
                popover.title = this.popoverTitle;
            if (this.popoverCloseOnClickOutside !== undefined)
                popover.closeOnClickOutside = this.popoverCloseOnClickOutside;
            if (this.popoverCloseOnMouseOutside !== undefined)
                popover.closeOnMouseOutside = this.popoverCloseOnMouseOutside;

            popover.onCloseFromOutside.subscribe(() => this.hide());
            if (this.popoverDismissTimeout > 0)
                setTimeout(() => this.hide(), this.popoverDismissTimeout);
            popover.show();
        }

        this.onShown.emit(this);
    }

    hide() {
        if (!this.visible) return;

        this.visible = false;
        if (this.popover)
            this.popover.destroy();

        if (this.content instanceof PopoverContentComponent)
            (this.content as PopoverContentComponent).hideFromPopover();

        this.onHidden.emit(this);
    }

    getElement() {
        return this.viewContainerRef.element.nativeElement;
    }

}
