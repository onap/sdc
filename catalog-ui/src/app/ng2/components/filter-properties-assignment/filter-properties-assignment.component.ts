/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

/**
 * Created by rc2122 on 5/16/2017.
 */
import {Component, Input, Output, EventEmitter, ViewChild} from '@angular/core';
import {ButtonModel, ButtonsModelMap, FilterPropertiesAssignmentData} from "app/models";
import {PopoverComponent} from "../popover/popover.component";
import * as sdcConfig from "../../../../../configurations/dev"

@Component({
    selector: 'filter-properties-assignment',
    templateUrl: './filter-properties-assignment.component.html',
    styleUrls: ['./filter-properties-assignment.component.less']
})

export class FilterPropertiesAssignmentComponent {
    @Input() componentType: string;
    @Output() searchProperties: EventEmitter<FilterPropertiesAssignmentData> = new EventEmitter<FilterPropertiesAssignmentData>();
    footerButtons:ButtonsModelMap = {};
    typesOptions:Array<string>;//All optional types
    selectedTypes:Object = {};
    allSelected:boolean = false;//if all option selected
    filterData:FilterPropertiesAssignmentData = new FilterPropertiesAssignmentData();
    @ViewChild('filterPopover') filterPopover: PopoverComponent;

    ngOnInit() {
        this.footerButtons['Apply'] = new ButtonModel('Apply', 'blue', this.search, this.someTypesSelectedAndThereIsPropertyName);
        this.footerButtons['Close'] = new ButtonModel('Close', 'grey', this.close);
        this.componentType = this.componentType.toLocaleLowerCase();
        this.typesOptions = sdcConfig.resourceTypesFilter[this.componentType];
    }

    selectAll = () => {
        _.forEach(this.typesOptions, (type) => {
            this.selectedTypes[type] = this.allSelected;
        });
    }

    onTypeSelected = (type:string) => {
        if(!this.selectedTypes[type]){
            this.allSelected = false;//unselected 'All'
        }
    };

    search = () => {
        console.log('search props');
        this.filterData.selectedTypes = [];
        _.forEach(sdcConfig.resourceTypesFilter[this.componentType], (type) => {
            if(this.selectedTypes[type]){
                this.filterData.selectedTypes.push(type);
            }
        });
        this.searchProperties.emit(this.filterData);
        this.filterPopover.hide();
    }

    close = () => {
        this.filterPopover.hide();
    }

    someTypesSelectedAndThereIsPropertyName = ():boolean => {
        if( _.find(Object.keys(this.selectedTypes),(key) => {
            return this.selectedTypes[key];
        }) && this.filterData.propertyName ){
            return null
        }
        return true;
    }

    clearAll = ():void => {
        this.filterData.propertyName = "";
        _.forEach(this.selectedTypes,(value, key) => {
            this.selectedTypes[key] = false;
        });
        this.allSelected = false;
    }

}
