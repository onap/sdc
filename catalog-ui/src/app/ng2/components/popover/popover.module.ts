/**
 * Created by rc2122 on 5/17/2017.
 */
import {NgModule} from "@angular/core";
import { CommonModule } from '@angular/common';
import {PopoverComponent} from "./popover.component";
import {PopoverContentComponent} from "./popover-content.component";

@NgModule({
    declarations: [
        PopoverComponent,
        PopoverContentComponent
    ],
    imports: [
        // PopoverComponent,
        // PopoverContentComponent
        CommonModule
    ],
    exports: [
        PopoverComponent,
        PopoverContentComponent
    ],
    providers: []
})
export class PopoverModule {

}
