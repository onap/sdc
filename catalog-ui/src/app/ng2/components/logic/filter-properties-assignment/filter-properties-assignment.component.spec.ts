import {async, ComponentFixture} from "@angular/core/testing";
import {FilterPropertiesAssignmentComponent} from "./filter-properties-assignment.component";
import {ConfigureFn, configureTests} from "../../../../../jest/test-config.helper";
import {NO_ERRORS_SCHEMA} from "@angular/core";
import {FilterPropertiesAssignmentData} from "../../../../models/filter-properties-assignment-data";
import {PopoverComponent} from "../../ui/popover/popover.component";



describe('filter-properties-assignemnt component', () => {

    let fixture: ComponentFixture<FilterPropertiesAssignmentComponent>;

    beforeEach(
        async(() => {

            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [FilterPropertiesAssignmentComponent],
                    imports: [],
                    schemas: [NO_ERRORS_SCHEMA],
                    providers: [],
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(FilterPropertiesAssignmentComponent);

            });
        })
    );


    it('should match current snapshot of artifact-tab component', () => {
        expect(fixture).toMatchSnapshot();
    });

    it('on selectAll', () => {
        let filterData:FilterPropertiesAssignmentData = new FilterPropertiesAssignmentData();
        filterData.propertyName = 'testVal';
        let typesOptions:Array<string> = ['option1', 'option2', 'option3'];
        let selectedTypes:Object = {};

        fixture.componentInstance.filterData = filterData;
        fixture.componentInstance.typesOptions = typesOptions;
        fixture.componentInstance.selectedTypes = selectedTypes;

        fixture.componentInstance.selectAll();

        let expectedRes = {"option1": false,"option2": false,"option3": false};
        expect(fixture.componentInstance.selectedTypes).toEqual(expectedRes);
    });


    it ('on onTypeSelected allSelected set to False', () => {
        let selectedTypes:Object = {"option1": true,"option2": false,"option3": true};
        fixture.componentInstance.selectedTypes = selectedTypes;
        fixture.componentInstance.allSelected = true;
        fixture.componentInstance.onTypeSelected('option2');

        expect(fixture.componentInstance.allSelected).toBe(false);
    });

    it ('on onTypeSelected allSelected remains True', () => {
        let selectedTypes:Object = {"option1": true,"option2": true,"option3": true};
        fixture.componentInstance.selectedTypes = selectedTypes;
        fixture.componentInstance.allSelected = true;
        fixture.componentInstance.onTypeSelected('option2');

        expect(fixture.componentInstance.allSelected).toBe(true);
    });

    it ('on clearAll', () => {
        let filterData:FilterPropertiesAssignmentData = new FilterPropertiesAssignmentData();
        filterData.propertyName = 'testVal';
        let selectedTypes:Object = {"option1": true,"option2": false,"option3": true};

        fixture.componentInstance.filterData = filterData;
        fixture.componentInstance.selectedTypes = selectedTypes;
        fixture.componentInstance.allSelected = true;

        fixture.componentInstance.clearAll();

        expect(fixture.componentInstance.filterData.propertyName).toBe('');
        expect(fixture.componentInstance.allSelected).toBe(false);
    });

    it ('someTypesSelectedAndThereIsPropertyName return True', ()=> {
        let res = fixture.componentInstance.someTypesSelectedAndThereIsPropertyName();

        expect(res).toBe(true)
    });

    it ('someTypesSelectedAndThereIsPropertyName return Null', ()=> {
        let selectedTypes:Object = {"option1": true,"option2": false,"option3": true};
        let filterData:FilterPropertiesAssignmentData = new FilterPropertiesAssignmentData();
        filterData.propertyName = 'testVal';

        fixture.componentInstance.selectedTypes = selectedTypes;
        fixture.componentInstance.filterData = filterData;

        let res = fixture.componentInstance.someTypesSelectedAndThereIsPropertyName();

        expect(res).toBe(null)
    });

    it ('search', ()=> {

        let filterData:FilterPropertiesAssignmentData = new FilterPropertiesAssignmentData();
        filterData.selectedTypes = ["CP"];
        fixture.componentInstance.filterData = filterData;

        let componentType: string = 'resource';
        fixture.componentInstance.componentType = componentType;

        let selectedTypes:Object = {"option1": true,"CP": true,"option3": true};
        fixture.componentInstance.selectedTypes = selectedTypes;

        let temp:any;
        let filterPopover: PopoverComponent = new PopoverComponent(temp , temp );
        fixture.componentInstance.filterPopover = filterPopover;
        fixture.componentInstance.filterPopover.hide = jest.fn();

        fixture.componentInstance.search();

        expect(fixture.componentInstance.filterData.selectedTypes).toEqual(["CP"]);
        expect(fixture.componentInstance.filterPopover.hide).toHaveBeenCalled();
    });

    it('close', () => {
        let temp:any;
        let filterPopover: PopoverComponent = new PopoverComponent(temp , temp );
        fixture.componentInstance.filterPopover = filterPopover;
        fixture.componentInstance.filterPopover.hide = jest.fn();

        fixture.componentInstance.close();

        expect(fixture.componentInstance.filterPopover.hide).toHaveBeenCalled();
    });

});