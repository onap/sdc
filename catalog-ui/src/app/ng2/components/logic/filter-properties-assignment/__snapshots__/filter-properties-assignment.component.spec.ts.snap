// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter-properties-assignemnt component should match current snapshot of artifact-tab component 1`] = `
<filter-properties-assignment
  allSelected="false"
  clearAll={[Function Function]}
  close={[Function Function]}
  filterData={[Function FilterPropertiesAssignmentData]}
  filterPopover={[Function ElementRef]}
  footerButtons={[Function Object]}
  onTypeSelected={[Function Function]}
  search={[Function Function]}
  searchProperties={[Function EventEmitter]}
  selectAll={[Function Function]}
  selectedTypes={[Function Object]}
  someTypesSelectedAndThereIsPropertyName={[Function Function]}
>
  <popover-content
    placement="bottom-right"
  >
    <form>
      <div
        class="field"
      >
        <label>
          Resource Type
        </label>
        <div>
          <checkbox
            data-tests-id="filter-checkbox-all"
          />
        </div>
        
      </div>
      <div
        class="field"
      >
        <label>
          Property Name
        </label>
        <input
          class="i-sdc-form-input"
          data-tests-id="filter-box"
          name="propertyName"
          placeholder="Type here"
          required=""
        />
      </div>
    </form>
  </popover-content><div
    class="open-filter-button"
  >
    <div
      class="sprite-new filter-icon"
      data-tests-id="filter-button"
    />
  </div>
</filter-properties-assignment>
`;
